purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/PurchasesIntegrationTest.kt

```diff
@@ -0,0 +1,190 @@
+                every { stateListener = any() } answers { latestStateListener = firstArg() }
+            }
+
+            proxyUrl?.let { urlString ->
+                Purchases.proxyURL = URL(urlString)
+            }
+
+            Purchases.configure(
+                PurchasesConfiguration.Builder(it, Constants.apiKey)
+                    .appUserID(testUserId)
+                    .build(),
+                mockBillingAbstract
+            )
+        }
+    }
+
+    @After
+    fun tearDown() {
+        Purchases.sharedInstance.close()
+    }
+
+    // region tests
+
+    @Test
+    fun sdkCanBeConfigured() {
+        onActivityReady {
+            assertThat(Purchases.sharedInstance.appUserID).isNotNull
+        }
+    }
+
+    @Test
+    fun customerInfoCanBeFetched() {
+        val lock = CountDownLatch(1)
+
+        onActivityReady {
+            Purchases.sharedInstance.getPurchaserInfoWith(
+                onError = { fail("should be success. Error: ${it.message}") },
+                onSuccess = { lock.countDown() }
+            )
+        }
+        lock.await(testTimeoutInSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    @Test
+    fun canFetchOfferings() {
+        val lock = CountDownLatch(1)
+
+        val productDetails = ProductDetailsFactory.createProductDetails()
+        mockBillingAbstract.mockQuerySkuDetails(querySkuDetailsSubsReturn = listOf(productDetails))
+
+        onActivityReady {
+            Purchases.sharedInstance.getOfferingsWith(
+                onError = { error -> fail("Get offerings should be successful. Error: ${error.message}") },
+                onSuccess = { offerings ->
+                    assertThat(offerings.current).isNotNull
+                    assertThat(offerings.current?.availablePackages?.size).isEqualTo(1)
+                    assertThat(offerings.current?.monthly?.product?.sku).isEqualTo(Constants.productIdToPurchase)
+
+                    lock.countDown()
+                }
+            )
+        }
+        lock.await(testTimeoutInSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    @Test
+    fun canPurchaseSubsProduct() {
+        val lock = CountDownLatch(1)
+
+        val productDetails = ProductDetailsFactory.createProductDetails()
+        val purchaseDetails = PurchaseDetailsFactory.createPurchaseDetails()
+        mockBillingAbstract.mockQuerySkuDetails(querySkuDetailsSubsReturn = listOf(productDetails))
+
+        onActivityReady { activity ->
+            Purchases.sharedInstance.purchaseProductWith(
+                activity = activity,
+                skuDetails = productDetails.skuDetails,
+                onError = { error, _ -> fail("Purchase should be successful. Error: ${error.message}") },
+                onSuccess = { _, customerInfo ->
+                    assertThat(customerInfo.allPurchaseDatesByProduct.containsKey(productDetails.sku)).isTrue
```

Comment:
should we check that the entitlement is there too? I mean, I get that that this gets indirectly calculated from the entitlement info, but still

Reply:
I'm waiting for the fix for that to be merged into the load shedder. Our test app uses the new product id format, which is causing entitlements not to be granted. Once that's fixed, I need to revisit integration tests for all 3 versions and add that test.